atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/lock/SimpleEteLockWatchResource.java

```diff
@@ -0,0 +1,160 @@
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import org.immutables.value.Value;
+
+public class SimpleEteLockWatchResource implements EteLockWatchResource {
+    public static final Namespace NAMESPACE = Namespace.create("lock");
+    private static final byte[] VALUE = PtBytes.toBytes("value");
+    private static final byte[] COLUMN = PtBytes.toBytes("b");
+
+    private final TransactionManager transactionManager;
+    private final ExposedLockWatchManager lockWatchManager;
+    private final Map<TransactionId, TransactionAndCondition> activeTransactions = new ConcurrentHashMap<>();
+
+    private String table = "watch";
+    private TableReference lockWatchTable = TableReference.create(NAMESPACE, table);
+
+    public SimpleEteLockWatchResource(TransactionManager transactionManager) {
+        this.transactionManager = transactionManager;
+        createTable();
+        this.lockWatchManager = new ExposedLockWatchManager(transactionManager.getLockWatchManager());
+    }
+
+    @Override
+    public TransactionId startTransaction() {
+        CommitUpdateCondition condition = new CommitUpdateCondition();
+        OpenTransaction transaction =
+                Iterables.getOnlyElement(transactionManager.startTransactions(ImmutableList.of(condition)));
+        TransactionId id = TransactionId.of(transaction.getTimestamp());
+        condition.setStartTs(transaction.getTimestamp());
+        activeTransactions.put(id, TransactionAndCondition.of(transaction, condition));
+        return id;
+    }
+
+    @Override
+    public Optional<CommitUpdate> endTransaction(TransactionId transactionId) {
+        TransactionAndCondition txnAndCondition = activeTransactions.remove(transactionId);
+        txnAndCondition.transaction().commit();
+        txnAndCondition.transaction().finish(unused -> null);
+        txnAndCondition.condition().cleanup();
+        return Optional.ofNullable(txnAndCondition.condition().getCommitUpdate());
+    }
+
+    @Override
+    public void write(WriteRequest writeRequest) {
+        Transaction transaction = activeTransactions.get(writeRequest.id()).transaction();
+        Map<Cell, byte[]> byteValues = getValueMap(writeRequest.rows());
+        transaction.put(lockWatchTable, byteValues);
+    }
+
+    @Override
+    public TransactionsLockWatchUpdate getUpdate(GetLockWatchUpdateRequest updateRequest) {
+        return lockWatchManager.getUpdateForTransactions(updateRequest.startTimestamps(), updateRequest.version());
+    }
+
+    @Override
+    public LockWatchVersion getVersion(TransactionId transactionId) {
+        return lockWatchManager
+                .getUpdateForTransactions(ImmutableSet.of(transactionId.startTs()), Optional.empty())
+                .startTsToSequence()
+                .get(transactionId.startTs());
+    }
+
+    @Override
+    public void setTable(String tableName) {
+        table = tableName.replace("\"", "");
```

Comment:
I don't know why, but I kept getting issues where the input table name was `"blah"` instead of `blah`.